lib/compress/index.js

```diff
@@ -1598,6 +1599,52 @@ def_optimize(AST_Switch, function(self, compressor) {
     while (i < len) eliminate_branch(self.body[i++], body[body.length - 1]);
     self.body = body;
 
+    // group equivalent branches so they will be located next to each other,
+    // that way the next micro-optimization will merge them.
+    // ** bail micro-optimization if not a simple switch case with breaks
+    if (body.every(branch =>
+        (
+            branch instanceof AST_Default || // default
+            !branch.expression.has_side_effects(compressor) // or no_side_effects
+        )
+            && (
+                branch.body.length === 0 // empty
+                || is_break(branch.body[branch.body.length - 1], compressor) // or has a break
+                || is_return(branch.body[branch.body.length - 1]) // or has a return
+            )
+    )) {
+        for (let i = 0; i < body.length; i++) {
+            let branch = body[i];
+            if (branch.body.length === 0) continue;
+            if (!aborts(branch)) continue;
+            for (let j = i + 1; j < body.length; j++) {
+                let next = body[j];
+                if (next.body.length === 0) continue;
+                if (
+                    branches_equivalent(next, branch, false)
+                    || (j === body.length - 1 && branches_equivalent(next, branch, true))
+                ) {
+                    // let's find previous siblings with empty fallthrough...
+                    let x = j-1;
+                    let fallthroughDepth = 0;
+                    while(x > i) {
+                        let previousSiblingBranch = body[x--];
+                        if(previousSiblingBranch.body.length === 0 && previousSiblingBranch instanceof AST_Case) {
```

Comment:
Why is the `AST_Case` necessary? Shouldn't `default` be allowed, too?

Reply:
@jridgewell  you are correct